### PR TITLE
Renaming of fsm args for better readability

### PR DESCRIPTION
**Pull request by HDykins**
Small naming change for readability, and typing change

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -101,7 +101,7 @@ export default class Machine extends erx.Bus<StateName> {
     return true;
   }
 
-  send(event: EventID, data?: any): boolean {
+  send(event: EventID, getState?: Function): boolean {
```
Comment (evenius) [inline src/index.ts]: Signatuuuure

File: src/index.ts [hunk 2]
```diff
@@ -60,7 +60,7 @@ export default class Machine extends erx.Bus<StateName> {
     this.transitioning = false;
   }
 
-  goTo(next: StateName, data?: any): boolean {
+  goTo(next: StateName, getState?: Function): boolean {
```
Comment (evenius) [inline src/index.ts]: Please add the function signature `() => state.MachineState<A>`

Comment (HDykins) [reply src/index.ts]: Hmmmm, the machineState lives on cyan under core/state/index.js. Shall i import from there? or move the types over to here? Or the whole thing!?

Comment (evenius) [reply src/index.ts]: Hmm, that is a really good point...

Comment (evenius) [reply src/index.ts]: Okay, how about `goTo(next: StateName, getState?: () => S)` and add the `FSM<S>` type in the class generic declaration.

And then, back in `src/core/index.ts`, you define FSM as `FSM<state.MachineState>`, and... Yea, same in `src/index.ts`

**Pull request merged**